website/content/docs/api-gateway/upgrades.mdx

```diff
@@ -9,6 +9,151 @@ description: >-
+  ```
+
+  You should receive a response similar to the following:
+
+  ```log
+  "hashicorp/consul-api-gateway:0.4.0"
+  ```
+
+  <a name="referencegrant"/>
+
+1. Issue the following command to get all `ReferencePolicy` resources across all namespaces. If you have any active `ReferencePolicy` resources, you will receive output similar to the response below.
+
+  ```shell-session
+  $ kubectl get referencepolicy --all-namespaces
+  Warning: ReferencePolicy has been renamed to ReferenceGrant. ReferencePolicy will be removed in v0.6.0 in favor of the identical ReferenceGrant resource.
+  NAMESPACE   NAME
+  default     example-reference-policy
+  ```
+
+  If your output is empty, proceed to upgrading your `GatewayClass`, `Gateway` and `HTTPRoute` resources to [v1beta1](#v1beta1-gatewayclass-gateway-httproute).
+
+1. For each `ReferencePolicy`, edit the source YAML file, updating to `kind: ReferenceGrant` and optionally updating the `metadata.name` field or filename if they previously included the term "policy", as in the example below. Note that because the `kind` field is being updated, it is not possible to use `kubectl edit` to edit the remote state directly.
+
+  <CodeBlockConfig hideClipboard filename="referencegrant.yaml">
+
+  ```yaml
+  apiVersion: gateway.networking.k8s.io/v1alpha2
+  kind: ReferenceGrant
+  metadata:
+    name: reference-grant
+    namespace: web-namespace
+  spec:
+    from:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      namespace: example-namesapce
+    to:
+    - group: ""
+      kind: Service
+      name: web-backend
+  ```
+
+  </CodeBlockConfig>
+
+1. For each file, apply the updated YAML to your cluster to create a new `ReferenceGrant` resource.
+
+  ```shell-session
+  $ kubectl apply -f <file>
+  ```
+
+1. Check to confirm that each new `ReferenceGrant` was created successfully.
+
+  ```shell-session
+  $ kubectl get referencegrant <name> --namespace <namespace>
+  NAME
+  example-reference-grant
+  ```
+
+1. Finally, delete each corresponding old `ReferencePolicy` resource. Because replacement `ReferenceGrant` resources have already been created, there should be no interruption in the availability of any referenced `Service` or `Secret`.
+
+  ```shell-session
+  $ kubectl delete referencepolicy <name> --namespace <namespace>
+  Warning: ReferencePolicy has been renamed to ReferenceGrant. ReferencePolicy will be removed in v0.6.0 in favor of the identical ReferenceGrant resource.
+  referencepolicy.gateway.networking.k8s.io "example-reference-policy" deleted
+  ```
+
+  <a name="v1beta1-gatewayclass-gateway-httproute"/>
+
+1. For each `GatewayClass`, `Gateway`, and `HTTPRoute`, edit the source YAML file, updating to `apiVersion: gateway.networking.k8s.io/v1beta1`. Note that because the `apiVersion` field is being updated, it is not possible to use `kubectl edit` to edit the remote state directly.
```

Comment:
```suggestion
1. For each `GatewayClass`, `Gateway`, and `HTTPRoute` object in the source YAML, update the `apiVersion` field to `gateway.networking.k8s.io/v1beta1`. Note that updating the `apiVersion` field prevents you from using the `kubectl edit` command to edit the remote state directly.
```
Now that I'm rereading that note about the `kubectl edit` command, I'm not sure that this is what we mean, either. Is it that certain fields (`apiVersion` and `kind`) are not editable or accessible from the command? The way it's worded, it seems as if the simple act of changing those fields is the reason why you can't use the command.

Reply:
> Is it that certain fields (apiVersion and kind) are not editable or accessible from the command?

Yes, that's correct, it will return an error and fail to update the remote resource.

Reply:
Removed the word `object` to align with suggested language for the ReferencePolicy update above.